semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/rpc/RpcSemanticConventionUtils.java

```diff
@@ -328,4 +357,66 @@ public static String getGrpcErrorMsg(Event event) {
 
     return Optional.empty();
   }
+
+  static Optional<String> getSanitizedGrpcAuthority(Event event) {
+
+    Optional<String> grpcAuthority = getGrpcAuthority(event);
+    if (grpcAuthority.isPresent()) {
+      return getSanitizedAuthorityValue(grpcAuthority.get());
+    } else if (SpanAttributeUtils.containsAttributeKey(
+        event, RawSpanConstants.getValue(Envoy.ENVOY_GRPC_AUTHORITY))) {
+      return getSanitizedAuthorityValue(
+          SpanAttributeUtils.getStringAttribute(
+              event, RawSpanConstants.getValue(Envoy.ENVOY_GRPC_AUTHORITY)));
+    }
+    return Optional.empty();
+  }
+
+  static Optional<String> getSanitizedAuthorityValue(String value) {
+    if (StringUtils.isBlank(value)) {
+      return Optional.empty();
+    }
+    // authority part of the uri assumes this format: <userinfo@host:port>
+    // parsing it as uri doesn't work, since this is not in url format
+    // allow for string of type <xyz> or <xyz:port> or <abc@xyz:port>
+    List<String> list = Splitter.on(COLON).splitToList(value);
```

Comment:
Shall `Splitter.on(COLON)` be initialized every time?

Reply:
name `list` is very generic and makes it hard to read in line 383. What is this list of? Can't we name it `uriPieces` or something like that?